internal/services/notification/db/migrate.go

```diff
@@ -69,3 +77,107 @@ func (d *DB) migrateV3(tx *sql.Tx) error {
 
 	return nil
 }
+
+func (d *DB) migrateV4(tx *sql.Tx) error {
+	var ddlPostgres = []string{
+		"alter table runwebhook add column project_id varchar",
+	}
+
+	var ddlSqlite3 = []string{
+		"create table new_runwebhook (id varchar NOT NULL, revision bigint NOT NULL, creation_time timestamp NOT NULL, update_time timestamp NOT NULL, payload blob NOT NULL, project_id varchar NOT NULL, PRIMARY KEY (id))",
+		"insert into new_runwebhook select *, '' from runwebhook",
+		"DROP TABLE runwebhook",
+		"ALTER TABLE new_runwebhook RENAME TO runwebhook",
+	}
+
+	var stmts []string
+	switch d.sdb.Type() {
+	case sql.Postgres:
+		stmts = ddlPostgres
+	case sql.Sqlite3:
+		stmts = ddlSqlite3
+	}
+
+	for _, stmt := range stmts {
+		if _, err := tx.Exec(stmt); err != nil {
+			return errors.WithStack(err)
+		}
+	}
+
+	runwebhookSb := sq.NewSelectBuilder().Select("id", "revision", "creation_time", "update_time", "payload").From("runwebhook").OrderBy("id").Limit(runWebhookLimit)
+
+	type runWebhookType struct {
+		ID           string
+		CreationTime time.Time
+		UpdateTime   time.Time
+		Revision     uint64
+		Payload      []byte
+		ProjectID    string
+	}
+
+	runWebhooks := make([]runWebhookType, 0)
+
+	var lastID string
+	hasNext := true
+	for hasNext {
+		rows, err := d.query(tx, runwebhookSb)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		count := 0
+		for rows.Next() {
+			var r runWebhookType
+			err = rows.Scan(&r.ID, &r.Revision, &r.CreationTime, &r.UpdateTime, &r.Payload)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+
+			var runWebhook types.RunWebhook
+			if err = json.Unmarshal(r.Payload, &runWebhook); err != nil {
+				return errors.WithStack(err)
+			}
+
+			r.ProjectID = runWebhook.ProjectInfo.ProjectID
+
+			runWebhooks = append(runWebhooks, r)
+			lastID = r.ID
+			count++
+		}
+		if err := rows.Err(); err != nil {
+			return errors.WithStack(err)
+		}
+
+		for _, r := range runWebhooks {
+			ub := sq.NewUpdateBuilder()
+			ub.Update("runwebhook").Set(ub.Assign("project_id", r.ProjectID)).Where(ub.E("id", r.ID))
+			if _, err := d.exec(tx, ub); err != nil {
+				return errors.WithStack(err)
+			}
+		}
+
+		hasNext = count == runWebhookLimit
+		runwebhookSb.Where(runwebhookSb.G("id", lastID))
```

Comment:
In the fixtures you have 20 runwebhooks, but runWebhookLimit is 25 so you aren't testing multiple batches. Instead of increasing the number of webhooks you could lower runWebhookLimit to something like 5 only for tests.